### ISSUE TITLE
micro:bit gives Java exception

### ISSUE THREAD

**Issue by OwenBrotherwood** [sandeepmistry/arduino-nRF5#139]
SOLUTION:
For micro:bit, ensure that one has selected Programmer CMSIS-DAP

With Examples->Basic->Blink, Java complained (edit: also with blank)
Softdevice: none
Upload (no programmer)
Windows 10 with "current Java" and Arduino IDE 1.8.2
micro:bit on serial 6 (mbed driver installed) (edit: Get Board Info gives Native serial port, can't obtain info)

No real ideas what to change just now: I look around the system and "go figure"
Edit:
Quite a few hits for the first line in the error, all pointing to incorrect setup of OS, which is possible in my setup
`cc.arduino.packages.uploaders.SerialUploader.uploadUsingProgrammer(SerialUploader.java:314)`

```
Archiving built core (caching) in: C:\Users\owen\AppData\Local\Temp\arduino_cache_980979\core\core_sandeepmistry_nRF5_BBCmicrobit_softdevice_none_191a960b5ee8a1bfe6344e316db1a0b2.a
c:/users/owen/appdata/local/arduino15/packages/sandeepmistry/tools/gcc-arm-none-eabi/5_2-2015q4/bin/../lib/gcc/arm-none-eabi/5.2.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .heap by 4 bytes

c:/users/owen/appdata/local/arduino15/packages/sandeepmistry/tools/gcc-arm-none-eabi/5_2-2015q4/bin/../lib/gcc/arm-none-eabi/5.2.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .stack_dummy by 4 bytes

c:/users/owen/appdata/local/arduino15/packages/sandeepmistry/tools/gcc-arm-none-eabi/5_2-2015q4/bin/../lib/gcc/arm-none-eabi/5.2.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .heap by 4 bytes

c:/users/owen/appdata/local/arduino15/packages/sandeepmistry/tools/gcc-arm-none-eabi/5_2-2015q4/bin/../lib/gcc/arm-none-eabi/5.2.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .stack_dummy by 4 bytes

c:/users/owen/appdata/local/arduino15/packages/sandeepmistry/tools/gcc-arm-none-eabi/5_2-2015q4/bin/../lib/gcc/arm-none-eabi/5.2.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .heap by 4 bytes

c:/users/owen/appdata/local/arduino15/packages/sandeepmistry/tools/gcc-arm-none-eabi/5_2-2015q4/bin/../lib/gcc/arm-none-eabi/5.2.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .stack_dummy by 4 bytes

Sketch uses 1596 bytes (0%) of program storage space. Maximum is 262144 bytes.
java.lang.NullPointerException
	at cc.arduino.packages.uploaders.SerialUploader.uploadUsingProgrammer(SerialUploader.java:314)
	at cc.arduino.packages.uploaders.SerialUploader.uploadUsingPreferences(SerialUploader.java:89)
	at cc.arduino.UploaderUtils.upload(UploaderUtils.java:78)
	at processing.app.SketchController.upload(SketchController.java:713)
	at processing.app.SketchController.exportApplet(SketchController.java:686)
	at processing.app.Editor$DefaultExportHandler.run(Editor.java:2135)
	at java.lang.Thread.run(Thread.java:745)

```

**Comment by sandeepmistry** [comment]
This is now fixed by @carlosperate's PR #128.

**Comment by OwenBrotherwood** [comment]
Thanks :)
